### PR TITLE
Backporting for LTS 2.222.4

### PR DESCRIPTION
**Pull request by olivergondza**
Backporting has started and the RC is scheduled for 2020-05-06.

Please have look at the changes proposed and rejected so we have a consensus of what goes in by 2020-05-06.

At large, I propose to get the Read-Only config page changes backported, but to leave the banner rework out.

```
Fixed
-----

JENKINS-62133		Minor     		2.235
	Configure screen: Resource Root URL validation error - 404 instance-identity
	regression
	https://issues.jenkins-ci.org/browse/JENKINS-62133

JENKINS-61812		Minor     		2.234
	Read-only f:password field always shows "N/A"
	https://issues.jenkins-ci.org/browse/JENKINS-61812

JENKINS-61711		Minor     		2.232
	Git plugin global config incorrectly reports git not in PATH
	https://issues.jenkins-ci.org/browse/JENKINS-61711

JENKINS-61694		Major     		2.230
	Groovy Hooks may run before or after JOB_CONFIG_ADAPTED
	https://issues.jenkins-ci.org/browse/JENKINS-61694

JENKINS-61660		Minor     		2.232
	Lack of spacing between setup wizard form errors
	regression
	https://issues.jenkins-ci.org/browse/JENKINS-61660

JENKINS-61409		Critical  		2.229
	Websocket connections crash if message size is greater than 64Kb
	non-trivial-lts-backporting
	https://issues.jenkins-ci.org/browse/JENKINS-61409

JENKINS-61321		Blocker   		2.224
	Pipeline Job configuration appears read-only for users with project-level permissions after update to Jenkins 2.223
	regression
	https://issues.jenkins-ci.org/browse/JENKINS-61321

JENKINS-61284		Minor     		2.224
	Grey bar below textarea in read only mode looks weird
	https://issues.jenkins-ci.org/browse/JENKINS-61284

JENKINS-61202		Minor     		2.223
	Read only view of job configuration page
	https://issues.jenkins-ci.org/browse/JENKINS-61202
		Caused https://issues.jenkins-ci.org/browse/JENKINS-61321
		Caused https://issues.jenkins-ci.org/browse/JENKINS-61284

Rejected
--------

JENKINS-62065		Minor     		2.235
	The bread crumb navigation on the top of the job is not responding after clicking Apply
	regression
	https://issues.jenkins-ci.org/browse/JENKINS-62065

JENKINS-61478		Minor     		2.230
	"Apply" (and similar) banners look really bad with larger header bar
	regression
	https://issues.jenkins-ci.org/browse/JENKINS-61478
		Caused https://issues.jenkins-ci.org/browse/JENKINS-62065

JENKINS-61479		Minor     		2.230
	java.lang.ClassCastException: java.lang.Long cannot be cast to java.util.Date
	https://issues.jenkins-ci.org/browse/JENKINS-61479

JENKINS-61841		Minor     		2.235
	PathRemover methods can throw enormous CompositeIOExceptions
	https://issues.jenkins-ci.org/browse/JENKINS-61841
```

### CODE REVIEW

**Comment by daniel-beck** [comment]
I guess we can close https://github.com/jenkinsci/jenkins/pull/4635 now?

**Review by daniel-beck — APPROVED** [review]
Inclusions are reasonable. Individual commits diffs look correct. remoting 4.2.1 is a prerequisite (https://github.com/jenkinsci/remoting/releases/tag/remoting-4.2.1), that seems reasonable as well.

https://github.com/jenkinsci/jenkins/pull/4710/commits/953d147a3f5e42c864e97d1b491e95ad632a3513 necessitated https://github.com/jenkinsci/jenkins/pull/4710/commits/a11dd2205a7ece67b9595de9379e7fd584979301, that's fine too. (Plus https://github.com/jenkinsci/jenkins/pull/4710/commits/cfbcf329656e2214830c83fb385720d0e382ee99 is useful too.)

----

Just FTR, some of these `lts-candidate` issues are labeled as such for the _upcoming_ LTS line, so should retain their label.

And finally, it would be nice if, now that we're doing PRs instead of (just) summary emails, the summary could be changed to GH flavored Markdown, so that links can be clicked.

**Review by res0nance — APPROVED** [review]
Looks reasonable

**Review by MarkEWaite — APPROVED** [review]

File: test/src/test/java/jenkins/agents/WebSocketAgentsTest.java [hunk 1]
```diff
@@ -99,9 +100,10 @@ public void smokes() throws Exception {
             ).stdout(System.out).start());
             r.waitOnline(s);
             assertEquals("response", s.getChannel().call(new DummyTask()));
+            assertNotNull(s.getChannel().call(new FatTask()));
             FreeStyleProject p = r.createFreeStyleProject();
             p.setAssignedNode(s);
-            p.getBuildersList().add(Functions.isWindows() ? new BatchFile("echo hello") : new Shell("echo hello"));
+            p.getBuildersList().add(Functions.isWindows() ? new BatchFile("echo hello") : new Shell("dd if=/dev/random count=1024 bs=200"));
```
Comment (jglick) [inline test/src/test/java/jenkins/agents/WebSocketAgentsTest.java:106]: You did not include all of #4635, specifically: https://github.com/jenkinsci/jenkins/pull/4635/commits/27626ccbd0d388d635bb73c7f444f7195045afae

**Review by jglick — DISMISSED** [review]

File: test/src/test/java/jenkins/agents/WebSocketAgentsTest.java (hunk 1 above)
Comment (olivergondza) [reply test/src/test/java/jenkins/agents/WebSocketAgentsTest.java]: Good catch, reverting the isolated backport and pulling the whole #4635.

**Comment by olivergondza** [comment]
The test failure as a flake.

@oleg-nenashev, do you plan to further review this? I will go ahead merging and deploying as soon as you are done

**Review by oleg-nenashev — APPROVED** [review]
Sorry, I did not mean to block it, just self-requested the review to review it if I have time.
The list of backports looks good to me, I do not think we are missing anything critical.

No strong opinion about backporting the SystemRead fixes, IMHO they are not necessary for an experimental feature which needs more patches in the incoming LTS versions. But it is fine since they are already included

**Pull request merged**